### PR TITLE
Support running on `schedule` and `workflow_dispatch`, some other small fixes

### PR DESCRIPTION
**Pull request by GaryGSC**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -123,11 +135,11 @@ function requestWithRetry (options) {
...
 }
 
-async function getRfcIfAlreadyCreated (linkToCommits) {
-  // linkToCommits includes commit hashes, so it happens to be the best way to identify duplicate RFCs
+async function getRfcIfAlreadyCreated (linkToWorkflowRun) {
+  // linkToWorkflowRun includes the runId, which is stable between workflow re-runs
+  // BTW, sequential scheduled runs aren't considered re-runs
   const tableName = 'change_request'
-  const githubBotUsernameInServicenow = 'githubac'
-  const sysparmQuery = `type=standard^sys_created_by=${githubBotUsernameInServicenow}^descriptionLIKE${linkToCommits}` // ^ corresponds to "and", LIKE corresponds to "contains"
```
Comment (jvisker) [inline index.js:130]: Why did the `githubBotUsernameInServicenow` go away?

Comment (GaryGSC) [reply index.js:130]: Good question. It didn't improve the query for existing RFCs beyond what we were already getting from using the link in the description, and it made the potentially-risky assumption that the org secrets (`STANDARD_CHANGE_SANDBOX_CLIENT_KEY` et al.) were associated with that user.

Comment (jvisker) [reply index.js:130]: So does that mean the RFC just doesn't show who made the change at this point? I don't necessarily disagree with the concept, just seems like a kind of big contract change.

Comment (GaryGSC) [reply index.js:130]: The bot user was never the `assigned_to` user on the RFC. We still set that to the Net ID of the person who performed the action on GitHub (the pusher, the person who triggered the `workflow_dispatch`, or the person who last edited the cron expression for the scheduled run).

**Review by joshgubler — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (joshgubler) [reply index.js:130]: If I'm reading these lines correctly, this doesn't change which user "created" the RFC. It simply uses a better link in the description and makes the lookup of an existing RFC less likely to break.

Comment (GaryGSC) [reply index.js:130]: @jvisker - Did that satisfy your question/concern?

**Pull request merged**